pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2164,6 +2164,28 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+	// Check if the backup is of the type virtual machine and then collect the rules and only then delete it.
+	if backup.Spec.BackupObjectType == resourcecollector.PxBackupObjectType_virtualMachine {
```

Comment:
can you replace this check with exposed function 
if IsBackupObjecyTypeVirtualMachine(backup) {
}

Reply:
@vsundarraj-px Made changes accordingly